Small fixes and an error explanation

JohannesNE opened the pull request:
Changed the explanation for the error when using pivot_longer() without a cols argument.

SaraTomiolo commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,339 @@
+  > In `pivot_wider()`, if no argument is given to `id_cols` it defaults to select all columns. However, `pivot_longer` has no default for `cols`, and will throw an error if no argument is given.
+
+We could select every column, with the tidyselect function `everything()`. This is not what want (..., but you should try it to see why!). 
+
```
Hi Johannes, great idea! 
Maybe we can add a code chunk were we give and example on how to use the function everything().
I think it would be helpful

JohsEnevoldsen replied:
Great, i added it in 117a1de

JohannesNE committed:
add (free) illustration of wide and long format

explain pivot_longer() error

Conflicts:
	content/material/2020-01-17-intermediate-dplyr.Rmd

JohannesNE committed:
add example with everything()

JohannesNE committed:
add extra example. tidying messy data.

JohsEnevoldsen commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
````diff
@@ -0,0 +1,389 @@
+```{r}
+storms_wide <- storms_wide %>% 
+   pivot_wider(names_from = name, values_from = mean_wind)  
+```
````
This example is a bit unintuitive, as the resulting wide table is very large, and contains mostly NA (for every storm that was not present in a given year).

lwjohnst86 replied:
You have any suggestions to improve it?

JohsEnevoldsen commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+
+The main problem in this table is that the variable names also inclue a value (week number). We want to pivot this wide dataframe into a longer one, with each row containg an id, a week number, a systolic and a diastolic blood pressure and a weigth. For this we use `pivot_longer()`.
+
+We do not want to include `id` in the reshape, so we select every column except this with `-id` as the first argument. We then seperate the variable names (e.g. 'bp_Dia_week1') into the actual variable name ( 'bp_Dia') and the value of the week variable ('1'). This is donw by setting the `names_sep` parameter to the string that seperates the variable name from the week value (here '_week'). Becau ...
```
I'm not too proud of this formulations, but I think it is a very important use case for the pivot functions. Generally pivoting is an abstraction that is very hard to explain, but quite quickly becomes second nature.

lwjohnst86 replied:
```suggestion
We do not want to include `id` in the reshape, so we select every column except this with `-id` as the first argument.
We then separate the variable names (e.g. 'bp_Dia_week1') into the actual variable name ('bp_Dia') and the value of the week variable ('1'). 

This is done by setting the `names_sep` parameter to the string that seperates the variable name from the week value (here '_week'). 
Because we split each variable name into two, we must set `names_to` for both. We do this with the character vector `c('.value', 'week')`. 
The special value `'.value'` indicates that the corresponding substring of the original variable name will itself become a variable name for the values in the new dataframe.
This makes a lot more sense when you look at the output of the function. 
The parameter `names_ptypes` converts the variable `week` from a character to an integer.
```

JohannesNE committed:
Merge remote-tracking branch 'SaraTomiolo/intermediate-dplyr'

JohannesNE committed:
rogue comma ;)

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+
+date: "1/17/2020"
+description: >
+  Learn to reshape data in long or wide form, manage factors and dates using the packages dplyr, tidyr, lubridate, forcats and stringr.
```
```suggestion
  Learn to reshape data in long or wide form and dates using the packages dplyr, tidyr, and lubridate.
```

lwjohnst86 replied:
@JohannesNE you can add these suggestions directly by using the button "Add suggestion to batch" or "Commit suggestion".

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+author: 
+- "Sara Tomiolo"
+- "Luke Johnston"
+
```
You'll have to update the file by synching with this branch (pull from "upstream" aka the main repo if you are familiar with doing that. If not, let me know)

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+
+### Prerequisites
+
+In this lesson, we assume a bit of familiarity with the basic functions of dplyr (especially `group_by()` and `summarize()`), which we covered in the [intro_dplyr lesson](https://au-cru.github.io/site/material/2019-11-15-intro-dplyr/) 
```
```suggestion
In this lesson, we assume a bit of familiarity with the basic functions of dplyr (especially `group_by()` and `summarize()`), which we covered in the [intro dplyr lesson](https://au-cru.github.io/site/material/2019-11-15-intro-dplyr/) 
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+
+### Objectives
+
+1. To learn to restructure your data between long and wide format with 
```
```suggestion
1. To know how to apply the `pivot_wider()` and `pivot_longer()` tidyr functions to restructure your data between long and wide formats (and vice versa). 
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+### Objectives
+
+1. To learn to restructure your data between long and wide format with 
+`pivot_longer()` and `pivot_wider()` functions from tidyr package.
```
```suggestion
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+
+1. To learn to restructure your data between long and wide format with 
+`pivot_longer()` and `pivot_wider()` functions from tidyr package.
+2. Learn to combine `group_by()` and `summarize()` function with `pivot_longer()` and `pivot_wider()` functions.
```
```suggestion
2. To use the `group_by()` and `summarize()` dplyr functions with the `pivot_longer()` and `pivot_wider()` functions to apply more powerful and useful wrangling techniques to your skills.
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+1. To learn to restructure your data between long and wide format with 
+`pivot_longer()` and `pivot_wider()` functions from tidyr package.
+2. Learn to combine `group_by()` and `summarize()` function with `pivot_longer()` and `pivot_wider()` functions.
+3. Manage dates with lubridate functions.
```
```suggestion
3. To apply common lubridate functions to easily wrangle pesky date variables.
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+
+1. To re-structure your data in long or wide format.
+2. To reshape data by using grouping, summarizing and reshaping dataframes.
+3. To reformat dates with lubridate
```
Let's just delete these lines.

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+
+Often we receive databases that have been created by others, and in which information is organized in a "messy" way. Some other times, we simply need to reorder the structure of the data to be able to carry out certain data analysis or data visualization that require a specific data structure. 
+
+The tidyverse, provides some very powerful packages (e.g. dplyr, tidyr) that can allow for complete reorganization of our data in a few lines of codes. Using these packages and their functions have several advantages:
```
```suggestion
The tidyverse provides some very powerful packages (e.g. dplyr, tidyr) that can allow for complete reorganization of our data in a few lines of codes. Using these packages and their functions have several advantages:
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
````diff
@@ -0,0 +1,389 @@
+library(lubridate)
+```
+
+### Prerequisites
````
```suggestion
## Prerequisites
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+In this lesson, we assume a bit of familiarity with the basic functions of dplyr (especially `group_by()` and `summarize()`), which we covered in the [intro_dplyr lesson](https://au-cru.github.io/site/material/2019-11-15-intro-dplyr/) 
+
+
+### Objectives
```
```suggestion
## Objectives
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+2. To reshape data by using grouping, summarizing and reshaping dataframes.
+3. To reformat dates with lubridate
+
+### Why learning to reformat and reshape data?
```
```suggestion
## Why learning to reformat and reshape data?
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+
+The illustrations below shows how the same information can be represented in a long and a wide format. Notice that the value (a-f) of each variable/key (x,y,z) can be read for each id (1, 2) in both formats.
+
+![](https://github.com/gadenbuie/tidyexplain/raw/master/images/static/png/original-dfs-tidy.png)
```
```suggestion
![Img: https://github.com/gadenbuie/tidyexplain](https://github.com/gadenbuie/tidyexplain/raw/master/images/static/png/original-dfs-tidy.png)
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+The illustrations below shows how the same information can be represented in a long and a wide format. Notice that the value (a-f) of each variable/key (x,y,z) can be read for each id (1, 2) in both formats.
+
+![](https://github.com/gadenbuie/tidyexplain/raw/master/images/static/png/original-dfs-tidy.png)
+*Img: https://github.com/gadenbuie/tidyexplain*
```
```suggestion
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+![](https://github.com/gadenbuie/tidyexplain/raw/master/images/static/png/original-dfs-tidy.png)
+*Img: https://github.com/gadenbuie/tidyexplain*
+
+The functions we want to use for these operations are `pivot_longer()` and `pivot_wider()` from the tidyr package (Those of you already familiar with this package may know these functions as `spread()` and `gather()`. Although these two functions still work, they are being slowly replaced by the `pivot_**` functions).
```
```suggestion
The functions we want to use for these operations are `pivot_longer()` and `pivot_wider()` from the tidyr package. Those of you already familiar with this package may know the older functions as `spread()` and `gather()`. Although these two functions still work, they are being slowly replaced by the `pivot_**` functions.
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+
+To show how these functions work, we are going to use the built-in dataframe `population`.
+
+#### Familiarize with the dataframe
```
```suggestion
### Familiarize ourselves with the dataframe
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
````diff
@@ -0,0 +1,389 @@
+```
+
+
+### `Pivot_wider()`
````
```suggestion
### `pivot_wider()`: From long to wide
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
````diff
@@ -0,0 +1,389 @@
+
+```{r}
+population_wide <- 
+  pivot_wider(population, names_from = country, values_from = population) 
````
```suggestion
  pivot_wider(population, names_from = country, values_from = population)
population_wide
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
````diff
@@ -0,0 +1,389 @@
+
+```{r}
+population_wide <- population %>% 
+  pivot_wider(names_from = country, values_from = population) 
````
```suggestion
  pivot_wider(names_from = country, values_from = population) 
population_wide
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
````diff
@@ -0,0 +1,389 @@
+
+```{r, eval FALSE}
+us_income_wide <- us_rent_income %>% 
+  #because we are only interested in income and not in rent, we will extract the variable of interest from our database
````
```suggestion
  # We are only interested in income and not in rent, so we will extract these variables of interest from our database
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+us_income_wide <- us_rent_income %>% 
+  #because we are only interested in income and not in rent, we will extract the variable of interest from our database
+  filter(variable == "income") %>% 
+  #we select only the variables of interest to remain in our dataframe
```
```suggestion
  # We select only the variables of interest to remain in our dataframe
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+  filter(variable == "income") %>% 
+  #we select only the variables of interest to remain in our dataframe
+  select(NAME, variable, estimate) %>% 
+  pivot_wider(names_from = ______, values_from = _____)
```
```suggestion
  pivot_wider(names_from = ______, values_from = _____)
us_income_wide
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
````diff
@@ -0,0 +1,389 @@
+  pivot_wider(names_from = ______, values_from = _____)
+```
+
+### `Pivot_longer()`
````
```suggestion
### `pivot_longer()`: From wide to long
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+
+We can also do the opposite: a shift from a dataframe in wide format to a dataframe in long format. 
+Let's consider the `population_wide` dataframe that we just created as the starting point. 
+Let's say that we want to reshape this data to a format in which all countries are listed under a single variable, e.g. `country_name`, and the population data are listed under a variable called `population_data`. For this operation, we need to use the function `pivot_longer()`
```
```suggestion
Let's say that we want to reshape this data to a format in which all countries are listed under a single variable, e.g. `country_name`, and the population data are listed under a variable called `population_data`. For this operation, we need to use the function `pivot_longer()`.
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+Let's consider the `population_wide` dataframe that we just created as the starting point. 
+Let's say that we want to reshape this data to a format in which all countries are listed under a single variable, e.g. `country_name`, and the population data are listed under a variable called `population_data`. For this operation, we need to use the function `pivot_longer()`
+
+In this case, we want to specify:
```
```suggestion
In this case, we want to specify:

```

lwjohnst86 replied:
Needs a new line for the list to render properly

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
````diff
@@ -0,0 +1,389 @@
+  pivot_longer(everything(), names_to = "Country", values_to = "Population")
+```
+
+This breaks the row-wise relation between year and population, and is clearly not what we are want. Instead, we specify that the variable `year` is not to be involved in the reshaping. Columns not involved in the reshape will be copied, to preserve the row-wise relations in the data. 
````
```suggestion
This breaks the row-wise relation between year and population, and is clearly not what we want. Instead, we specify that the variable `year` is not to be involved in the reshaping. Columns not involved in the reshape will be copied, to preserve the row-wise relations in the data. 
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+
+  > In `pivot_wider()`, if no argument is given to `id_cols` it defaults to select all columns. However, `pivot_longer` has no default for `cols`, and will throw an error if no argument is given.
+
+We could select every column, with the [tidyselect](https://r4ds.had.co.nz/transform.html#select) helper function `everything()`. 
```
```suggestion
We could select every column, with the [tidyselect](https://r4ds.had.co.nz/transform.html#select) helper function `everything()`. This function tells `pivot_longer()` to use all the columns in the dataset when converting to long form.
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
````diff
@@ -0,0 +1,389 @@
+population_wide %>%
+  pivot_longer(-year, names_to = "country_name", values_to = "population_data")
+```
+
````
```suggestion

We use the `-` here for `year` to tell `pivot_longer()` to *not* include (to *exclude*) the column `year` from being converted to long form.
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+**Notice that** in the function `pivot_longer()` the arguments `names_to` and `values_to` require "" because we are providing names of variables that need to be created. 
+Instead, in the function `pivot_wider()` we are giving the arguments `names_from` and `values_from` variables that already exist, hence  the "" are not needed.
+
+#### Exercise
```
```suggestion
#### Exercise 2
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+  select(-GEOID) %>% 
+  #remember that the variable that will not be reshaped need to be omitted using the operator -
+  #when there is more than one variable to omit, we combined the operators - with c()
+  pivot_longer(-c(______, ______), names_to = ________, values_to = _________)
```
```suggestion
  pivot_longer(-c(______, ______), names_to = ________, values_to = _________)
us_rent_income_long
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
````diff
@@ -0,0 +1,389 @@
+
+```{r, eval = FALSE}
+us_rent_income_long <- us_rent_income %>% 
+  #this variable is a nuisance and is not giving information useful to us, so we remove it 
````
```suggestion
  # this variable is a nuisance and is not giving information useful to us, so we will remove it 
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+us_rent_income_long <- us_rent_income %>% 
+  #this variable is a nuisance and is not giving information useful to us, so we remove it 
+  select(-GEOID) %>% 
+  #remember that the variable that will not be reshaped need to be omitted using the operator -
```
```suggestion
  # Recall that the to exclude a variable from reshaping, you use `-` (minus symbol).
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+  #this variable is a nuisance and is not giving information useful to us, so we remove it 
+  select(-GEOID) %>% 
+  #remember that the variable that will not be reshaped need to be omitted using the operator -
+  #when there is more than one variable to omit, we combined the operators - with c()
```
```suggestion
  # When we want to indicate multiple variables, we can combine them with `c()`.
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+
+
+
+### Use `pivot_**` functions in combination with `group_by()` and `summarize()` functions
```
```suggestion
### `pivot_**` with `group_by()` and `summarize()`
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
````diff
@@ -0,0 +1,389 @@
+```
+
+
+### Handle dates with lubridate
````
We could even put this as an optional section, since the wrangling may actually take a while to go through and explain. (I know for me this ong-wide, wide-long conversion took a while to sink in and conceptualize).

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
````diff
@@ -0,0 +1,389 @@
+
+Once again, explore the data using the function `View()`
+
+```{r}
````
```suggestion
```{r, eval=FALSE}
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+### Use `pivot_**` functions in combination with `group_by()` and `summarize()` functions
+
+For this example we are going to use the built-in dataframe `storms`
+
```
```suggestion
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+Take another look at the `storms` dataframe. Variables are measured multiple times within each year. 
+So if we want to have a mean value of wind speed for each year, we need to use the functions `group_by()` and `summarize()` that we showed in the intro session. 
+
+We are going to call this new dataframe `storm_wide`
```
```suggestion
We are going to call this new dataframe `storm_wide`.
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+We are going to call this new dataframe `storm_wide`
+
+First we will calculate the mean value of wind speed for each storm, across years 
+and discriminating across storm status
```
```suggestion
and discriminating across storm status:
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
````diff
@@ -0,0 +1,389 @@
+
+```{r}
+storms_wide <- storms %>% 
+  #first we select the variable of interest to avoid an overcrowded dataframe
````
```suggestion
  # First we select the variables we are most interested in
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+storms_wide <- storms %>% 
+  #first we select the variable of interest to avoid an overcrowded dataframe
+  select(name, year, wind, status) %>% 
+  #second we tell R according to which criteria we want to summarize our data, i.e. we want to know the mean value of wind speed for each storm according to storm status and year
```
```suggestion
  # Next, we tell R how we want to summarize our data, i.e. we want
  # to know the mean wind speed for each storm by storm status and year
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
````diff
@@ -0,0 +1,389 @@
+  summarize(mean_wind = mean(wind)) 
+```
+
+Now we can reshape the data into a wide format 
````
```suggestion
Now we can reshape the data into a wide format:
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
````diff
@@ -0,0 +1,389 @@
+   pivot_wider(names_from = name, values_from = mean_wind)  
+```
+
+#### Exercise
````
```suggestion
#### Exercise 3
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+
+#### Exercise
+Using the dataframe `us_rent_income` use the functions `group_by()` and `summarize()` in combination with
+`pivot_longer()`, `pivot_wider()` to reshape a database in which different values of cut correspond to mean values of x, y, z (tip: you need to use both pivot_longer and pivot_wider!)
```
```suggestion
`pivot_longer()`, `pivot_wider()` to reshape a database in which different values of cut correspond to mean values of x, y, z (tip: you need to use both `pivot_longer()` and `pivot_wider()`!).
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+  pivot_ ______(____, names_ ___ = __________, values_ ___ = ___________) %>% 
+  pivot_ ________(names_ ____ = ________, values_ ____ = __________)
+  
+diamonds_reshaped
```
The exercise is not synched with the text for the exercise.

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+### Handle dates with lubridate
+
+Often enough, data are collected on the same subjects (in this case storm names) across time, this implied the use of dates. However, handling dates can become a bit of a headache if R does not recognize the format. 
+The package lubridate helps specifically with this problem and allows to format dates in a way that can be unequivocally recognized by R.
```
```suggestion
The package lubridate helps specifically with this problem and allows you format dates in a way that can be unequivocally recognized by R.
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
````diff
@@ -0,0 +1,389 @@
+library(lubridate)
+```
+
+The best way to input dates (if you are the one doing the data input), is to create three separate columns for day, month, year.
````
```suggestion
The best way to input dates (if you are the one doing the data input), is to create three separate columns for day, month, and year.
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+
+The best way to input dates (if you are the one doing the data input), is to create three separate columns for day, month, year.
+
+Then we can use the function `ymd()` from the package lubridate in combination with the function `paste()` to create a date variable
```
```suggestion
Then we can use the function `ymd()` (for "year", "month", "day") from the package lubridate in combination with the function `paste()` to create a date variable:
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
````diff
@@ -0,0 +1,389 @@
+
+```{r}
+storms_with_date <- storms %>% 
+  mutate(date = ymd(paste(year, month, day, sep = "-") )) 
````
```suggestion
  mutate(date = ymd(paste(year, month, day, sep = "-"))) 
storms_with_date
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
````diff
@@ -0,0 +1,389 @@
+  mutate(date = ymd(paste(year, month, day, sep = "-") )) 
+```
+
+if you query the new dataframe using the function `View()` you can see that the new variable date is specifically formatted as a Date data type.
````
```suggestion
If you query the new dataframe using the function `View()` you can see that the new variable date is specifically formatted as a Date data type.
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
````diff
@@ -0,0 +1,389 @@
+str(storms_with_date)
+```
+
+# Extra: Tidying messing data
````
```suggestion
## Extra: Tidying messing data
```

lwjohnst86 replied:
We could move this up and replace with the lubridate section. Then later have a session on lubridate and stringr or something.

lwjohnst86 replied:
I like this section. You should definitely do it at the end of the session to show off how powerful the pivot functions are.

JohsEnevoldsen replied:
Moved in section in df6cbb5653462c38561065d3fc78e5de4ec4ec3b

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
```diff
@@ -0,0 +1,389 @@
+
+# Extra: Tidying messing data
+
+So far, we have only reshaped "clean" datasets, but at major usecase for the `pivot_**` functions is tidying up messy data. In the example below, we will tidy a made up (but very realistic) table.
```
```suggestion
So far, we have only reshaped "clean" datasets, but a major use case for the `pivot_**` functions is tidying up messy data. In the example below, we will tidy a made up (but very realistic) table.
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
````diff
@@ -0,0 +1,389 @@
+
+```
+
+The main problem in this table is that the variable names also inclue a value (week number). We want to pivot this wide dataframe into a longer one, with each row containg an id, a week number, a systolic and a diastolic blood pressure and a weigth. For this we use `pivot_longer()`.
````
```suggestion
The main problem in this dataset is that the variable names also include data (week number). We want to pivot this wide dataframe into a longer one, with each row containg an id, a week number, a systolic and a diastolic blood pressure and a weight. For this we use `pivot_longer()`.
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
````diff
@@ -0,0 +1,389 @@
+
+```{r}
+blood_pressure_long <- blood_pressure %>% 
+  pivot_longer(-id, names_sep = '_week', names_to = c('.value', 'week'), names_ptypes = list(week = integer()))
````
```suggestion
  pivot_longer(
    -id, 
    names_sep = '_week', 
    names_to = c('.value', 'week'), 
    names_ptypes = list(week = integer())
  )
```

lwjohnst86 commented on content/material/2020-01-17-intermediate-dplyr.Rmd:
````diff
@@ -0,0 +1,389 @@
+blood_pressure_long
+```
+
+In some cases you may want only on column with blood pressures, and a seperate column indicating if the pressue is diastolic or systolic. We could pivot our `blood_pressure_long` even longer, but for the sake of demonstration, we will do it in one function call. As `weight` cannot be seperated in diastolic and systolic, we will first exclude this variable. 
````
```suggestion
In some cases you may want only one column with blood pressures, and a separate column indicating if the pressure is diastolic or systolic. 
We could pivot our `blood_pressure_long` even longer, but for the sake of demonstration, we will do it in one function call. 
Since `weight` cannot be separated in diastolic and systolic, we will first exclude this variable. 
```

lwjohnst86 reviewed (changes requested):
Looking good @SaraTomiolo and @JohannesNE! I've added several suggestions and changes. @JohannesNE since this is your PR, you'll have to "Commit suggestions" so it updates on your branch.

The final exercise text and code needs to be synchronized.

lwjohnst86 reviewed (changes requested):
Looking good @SaraTomiolo and @JohannesNE! I've added several suggestions and changes. @JohannesNE since this is your PR, you'll have to "Commit suggestions" so it updates on your branch.

The final exercise text and code needs to be synchronized.

JohannesNE committed:
Merge remote-tracking branch 'au-cru/master'

JohannesNE committed:
add source to img alt text

Co-Authored-By: Luke W Johnston <lwjohnst86@users.noreply.github.com>

JohannesNE committed:
remove img label

Co-Authored-By: Luke W Johnston <lwjohnst86@users.noreply.github.com>

JohannesNE committed:
better wording

Co-Authored-By: Luke W Johnston <lwjohnst86@users.noreply.github.com>

JohannesNE committed:
split long line

Co-Authored-By: Luke W Johnston <lwjohnst86@users.noreply.github.com>

JohannesNE committed:
tidy function call

Co-Authored-By: Luke W Johnston <lwjohnst86@users.noreply.github.com>

JohannesNE committed:
add short explanation of everything()

Co-Authored-By: Luke W Johnston <lwjohnst86@users.noreply.github.com>

JohannesNE committed:
fix typos

Co-Authored-By: Luke W Johnston <lwjohnst86@users.noreply.github.com>

JohannesNE committed:
Move 'tidying messy data'

JohannesNE committed:
fix typo

JohsEnevoldsen commented:
@lwjohnst86 I believe I have done this now.

JohannesNE committed:
add exercise on tidying messy data.

JohsEnevoldsen commented:
@lwjohnst86 I committed the suggestions that showed up when in the 'files changed' tab, but overlooked the suggestions in the thread above. These are marked as 'outdated'. I'm not sure why, or how to resolve this.

lwjohnst86 commented:
Thanks @JohannesNE and @SaraTomiolo! Great job :) Merged it in
